### PR TITLE
Website: speed up load time, enable service worker

### PR DESCRIPTION
**Pull request by azz**
More improvements to the website:

**Performance:**

* Add a service worker to pre-cache and allow offline website use.
* Load CodeMirror from a CDN to drastically improve initial load.
* Automatically load the parser set in the URL in the webworker immediately, rather than after CodeMirror loads. 

**Misc:**
* Add AST viewer.
* Add web app manifest.

Preview: https://azz.github.io/prettier/

### CODE REVIEW

**Comment by vjeux** [comment]
<3

**Comment by lydell** [comment]
In Firefox, I get this:

```
TypeError: setting getter-only property "name"  index.js:1257:5
```

No stack trace, unfortunately.

Chrome seems to work fine. Is this just me, or do you get that error in Firefox as well?

File: package.json [hunk 1]
```diff
@@ -48,6 +48,9 @@
     "rollup-plugin-node-globals": "1.1.0",
     "rollup-plugin-node-resolve": "2.0.0",
     "rollup-plugin-replace": "1.1.1",
+    "sw-toolbox": "3.6.0",
+    "typescript": "2.4.0",
+    "typescript-eslint-parser": "git://github.com/eslint/typescript-eslint-parser.git#479b592c0ad84a6ec5170fbe1821ed5ca90c4ee1",
```
Comment (vjeux) [inline package.json]: Why did you copy and paste those two here?

**Comment by vjeux** [comment]
If you fix the issue with firefox and address the package.json dependencies, feel free to merge!

**Comment by azz** [comment]
> In Firefox, I get this:
> TypeError: setting getter-only property "name"  index.js:1257:5

Can't reproduce in Firefox 47, 53 or 54 on Windows. :/

index.js:1257:5 seems to be the start of the entire package.json:

<details>
<pre>
var name = "prettier";
var version$1 = "1.4.4";
var description = "Prettier is an opinionated JavaScript formatter";
var bin = { "prettier": "./bin/prettier.js" };
var repository = "prettier/prettier";
var author = "James Long";
var license = "MIT";
var main = "./index.js";
var dependencies = {};
var devDependencies = { "babel-cli": "6.24.1", "babel-code-frame": "7.0.0-alpha.12", "babel-preset-es2015": "6.24.1", "babylon": "7.0.0-beta.10", "chalk": "1.1.3", "cross-spawn": "5.1.0", "diff": "3.2.0", "eslint": "3.19.0", "eslint-plugin-prettier": "2.1.1", "esutils": "2.0.2", "flow-parser": "0.47.0", "get-stdin": "5.0.1", "glob": "7.1.2", "graphql": "0.10.1", "jest": "20.0.0", "jest-validate": "20.0.3", "minimist": "1.2.0", "mkdirp": "^0.5.1", "parse5": "3.0.2", "postcss": "^6.0.1", "postcss-less": "^1.0.0", "postcss-media-query-parser": "0.2.3", "postcss-scss": "1.0.0", "postcss-selector-parser": "2.2.3", "postcss-values-parser": "git://github.com/shellscape/postcss-values-parser.git#5e351360479116f3fe309602cdd15b0a233bc29f", "prettier": "1.4.2", "rimraf": "2.6.1", "rollup": "0.41.1", "rollup-plugin-commonjs": "7.0.0", "rollup-plugin-json": "2.1.0", "rollup-plugin-node-builtins": "2.0.0", "rollup-plugin-node-globals": "1.1.0", "rollup-plugin-node-resolve": "2.0.0", "rollup-plugin-replace": "1.1.1", "typescript": "2.3.4", "typescript-eslint-parser": "git://github.com/eslint/typescript-eslint-parser.git#806251524424e3ad91e750da0f76b8de25ed0b42", "uglify-es": "3.0.15", "webpack": "2.6.1" };
var scripts = { "test": "jest", "test-integration": "jest tests_integration", "lint": "eslint .", "build": "./scripts/build/build.sh" };
var jest = { "setupFiles": ["<rootDir>/tests_config/run_spec.js"], "snapshotSerializers": ["<rootDir>/tests_config/raw-serializer.js"], "testRegex": "jsfmt\\.spec\\.js$|__tests__/.*\\.js$", "testPathIgnorePatterns": ["tests/new_react", "tests/more_react"] };
var _package = {
  name: name,
  version: version$1,
  description: description,
  bin: bin,
  repository: repository,
  author: author,
  license: license,
  main: main,
  dependencies: dependencies,
  devDependencies: devDependencies,
  scripts: scripts,
  jest: jest
};

var _package$1 = Object.freeze({
  name: name,
  version: version$1,
  description: description,
  bin: bin,
  repository: repository,
  author: author,
  license: license,
  main: main,
  dependencies: dependencies,
  devDependencies: devDependencies,
  scripts: scripts,
  jest: jest,
  default: _package
});
</pre>
</details>

Which would be trying to set readonly `window.name` to `"prettier"`.

Anyone know how this code gets in the `index.js` bundle?

**Comment by azz** [comment]
Looks like a consequence of the `require("./package.json").version`. Not sure why rollup is bundling it the way it is.

**Comment by azz** [comment]
@lydell Can you confirm that you also get this error on https://prettier.github.io/prettier/?

File: package.json (hunk 1 above)
Comment (azz) [inline package.json]: Bad "take both" merge resolution. Fixed.

**Comment by lydell** [comment]
Yes, it happens on https://prettier.github.io/prettier/ as well. But only in Firefox Developer Edition 55.0b2 (64-bit) (which is my main browser), not in Firefox 54.0 (64-bit) (which is the one that comes with Ubuntu 16.04). (I tested with completely new Firefox profiles.) It also happens on Firefox Nightly 56.0a1 (2017-06-17) (64-bit). So perhaps the fault is on Firefox's side?

Since it happens even without this PR, I guess we can merge this?

**Comment by azz** [comment]
Website now works offline! :tada:

**Pull request merged**